core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -706,8 +708,18 @@ angular.module('oppia').directive('questionsList', [
           $scope.$on(EVENT_QUESTION_SUMMARIES_INITIALIZED, function(ev) {
             _initTab(false);
           });
-
-          _initTab(true);
+          ctrl.$onInit = function() {
+            ctrl.skillIds = [];
+            ctrl.selectedSkillIds = [];
+            ctrl.associatedSkillSummaries = [];
+            ctrl.selectedSkillId = ctrl.getSelectedSkillId();
+            ctrl.getQuestionSummariesForOneSkill =
+              QuestionsListService.getCachedQuestionSummaries;
+            ctrl.getCurrentPageNumber = (
+              QuestionsListService.getCurrentPageNumber);
+            ctrl.editorIsOpen = false;
+            _initTab(true);
```

Comment:
Ditto -- inline? Similarly elsewhere in other files.

Reply:
Added comment.